### PR TITLE
Updates to Cluster API

### PR DESCRIPTION
**Pull request by taroface**
Fixes https://github.com/cockroachdb/docs/issues/11267.

@itsbilal , @mikeCRL  - There are a several TODOs included in the updated JSON. Should I manually remove these? Note that the description for `GCPolicy` only has a TODO item and no description.

### CODE REVIEW

**Comment by taroface** [comment]
@itsbilal @mikeCRL Friendly ping for review!

**Review by itsbilal — APPROVED** [review]
@taroface Sorry for missing this earlier!

And yes, the TODOs should be manually removed, sorry for that. Otherwise, this looks good to me!

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/12323#-:-Mpw-jZd2ElWtVr2yGvf:buvk2tw)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @mikeCRL and @taroface)

___
*[api/api-spec.json, line 1516 at r1](https://reviewable.io/reviews/cockroachdb/docs/12323#-Mpw-zOy47hKWPN2nQA4:-Mpw-zOy47hKWPN2nQA5:b-jmlkzb) ([raw file](https://github.com/cockroachdb/docs/blob/695ab35187595cadd438f5ec491a43fc4d7e222f/api/api-spec.json#L1516)):*
> ```JSON
>     },
>     "ZoneConfig": {
>       "description": "Note: when adding/removing fields here, be sure to update\nsupportedZoneConfigOptions in the sql package, to synchronize\nwith the code for ALTER ... CONFIGURE ZONE.",
> ```

This might also need to be removed, as it's just a code comment that seems out of place in docs.
___
*[api/api-spec.json, line 1521 at r1](https://reviewable.io/reviews/cockroachdb/docs/12323#-Mpw0GifCSANkA1Hucv0:-Mpw0GigD4tbcfsu1Ja_:b-uqe29e) ([raw file](https://github.com/cockroachdb/docs/blob/695ab35187595cadd438f5ec491a43fc4d7e222f/api/api-spec.json#L1521)):*
> ```JSON
>       "properties": {
>         "constraints": {
>           "description": "Constrains which stores the replicas can be stored on. The\norder in which the constraints are stored is arbitrary and may change.\nhttps://github.com/cockroachdb/cockroach/blob/master/docs/RFCS/20160706_expressive_zone_config.md#constraint-system\n\nNOTE: The sum of the num_replicas fields of the Constraints must add up to\nat most ZoneConfig.num_replicas, or there must be no more than a single\nConstraints field with num_replicas set to 0.",
> ```

The last sentence can also be removed. In general, I think a lot of the properties in this object can have no descriptions, and can be prefaced with "For internal use only" in the outer object's description above.

**Review by taroface — COMMENTED** [review]
Done, thanks @itsbilal !

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/12323#-:-Msf19783Lj7nzizPpbM:b-r2oh83)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @itsbilal and @mikeCRL)

___
*[api/api-spec.json, line 1516 at r1](https://reviewable.io/reviews/cockroachdb/docs/12323#-Mpw-zOy47hKWPN2nQA4:-Msf-NFr09Atx6qNW6eG:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/695ab35187595cadd438f5ec491a43fc4d7e222f/api/api-spec.json#L1516)):*
<details><summary><i>Previously, itsbilal (Bilal Akhtar) wrote…</i></summary><blockquote>

This might also need to be removed, as it's just a code comment that seems out of place in docs.
</blockquote></details>

Done.
___
*[api/api-spec.json, line 1521 at r1](https://reviewable.io/reviews/cockroachdb/docs/12323#-Mpw0GifCSANkA1Hucv0:-Msf-lOh8RSXr-EhTlH2:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/695ab35187595cadd438f5ec491a43fc4d7e222f/api/api-spec.json#L1521)):*
<details><summary><i>Previously, itsbilal (Bilal Akhtar) wrote…</i></summary><blockquote>

The last sentence can also be removed. In general, I think a lot of the properties in this object can have no descriptions, and can be prefaced with "For internal use only" in the outer object's description above.
</blockquote></details>

Done.

**Comment by taroface** [comment]
@kathancox, let me know if you have any questions about what I've done in this PR.

File: api/api-spec.json [hunk 1]
```diff
@@ -931,6 +1511,201 @@
...
+      "title": "Response for databaseTables.",
+      "properties": {
+        "next": {
+          "description": "The continuation token, for use in the next paginated call in the `offset`\nparameter.",
+          "type": "integer",
+          "format": "int64",
+          "x-go-name": "Next"
+        },
+        "table_names": {
+          "description": "The names of all tables in this database. Note that\nall responses will be schema-qualified (schema.table) and that every schema\nor table that contains a \"sql unsafe character\" such as uppercase letters\nor dots will be surrounded with double quotes, such as \"naughty schema\".table.",
```
Comment (kathancox) [inline api/api-spec.json:1659]: Same comment as above a few comments, if relevant

**Review by kathancox — APPROVED** [review]
LGTM! Some comments that are all minor. Hope this covers what you needed!

**Review by taroface — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/12323)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @itsbilal, @kathancox, and @mikeCRL)

___
*[api/api-spec.json, line 190 at r2](https://reviewable.io/reviews/cockroachdb/docs/12323#-MoeE1L2SOmTzNrDoeGj-r2-190:-Mt55e8N2s-S3vwEAJZf:b-9clxjg) ([raw file](https://github.com/cockroachdb/docs/blob/e17b7956bb93cc3c3a3de8ac5b44594d63e68e10/api/api-spec.json#L190)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

I might be misreading this, so disregard if it doesn't make sense! But, since this description talks about periods or uppercase letters as unsafe characters, I was looking for that within the double-quoted example. 
</blockquote></details>

Good point. @itsbilal could you advise on whether the example needs to be updated?
___
*[api/api-spec.json, line 288 at r2](https://reviewable.io/reviews/cockroachdb/docs/12323#-MoeE1L2SOmTzNrDoeGj-r2-288:-Mt58_JU-GHNwS3c9RBh:b-jft39n) ([raw file](https://github.com/cockroachdb/docs/blob/e17b7956bb93cc3c3a3de8ac5b44594d63e68e10/api/api-spec.json#L288)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

There are double double quotes here, is that right?
</blockquote></details>

Likely not! Fixed.
___
*[api/api-spec.json, line 399 at r2](https://reviewable.io/reviews/cockroachdb/docs/12323#-MoeE1L2SOmTzNrDoeGj-r2-399:-Mt58gDCDnBAT1XWKnWo:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/e17b7956bb93cc3c3a3de8ac5b44594d63e68e10/api/api-spec.json#L399)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

Maybe drop the hyphen here on "logged-in" and then that will be consistent with line 452 as well.
</blockquote></details>

Done.
___
*[api/api-spec.json, line 495 at r2](https://reviewable.io/reviews/cockroachdb/docs/12323#-MoeE1L2SOmTzNrDoeGj-r2-495:-Mt58hEl1Fa4p5ARMJqe:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/e17b7956bb93cc3c3a3de8ac5b44594d63e68e10/api/api-spec.json#L495)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

Ditto, on dropping the hyphen
</blockquote></details>

Done.
___
*[api/api-spec.json, line 1157 at r2](https://reviewable.io/reviews/cockroachdb/docs/12323#-MoeE1L2SOmTzNrDoeGj-r2-1157:-Mt58qbE-KdZAtSILFqn:bepwqjv) ([raw file](https://github.com/cockroachdb/docs/blob/e17b7956bb93cc3c3a3de8ac5b44594d63e68e10/api/api-spec.json#L1157)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

This is really nit-picky and I don't even though if this is true/right, but should "Subzones" on this line be singular to match the "Subzone" object?
</blockquote></details>

Not sure. @itsbilal can you confirm whether "Subzone" should be singular here?
___
*[api/api-spec.json, line 1187 at r2](https://reviewable.io/reviews/cockroachdb/docs/12323#-MoeE1L2SOmTzNrDoeGj-r2-1187:-Mt59CXs5Xo-bSHomMAo:b-ec1v8d) ([raw file](https://github.com/cockroachdb/docs/blob/e17b7956bb93cc3c3a3de8ac5b44594d63e68e10/api/api-spec.json#L1187)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

Wondering if 'target_id' should be surrounded by double quotes — I think that has been the pattern throughout
</blockquote></details>

I think these are probably meant to be code backticks, per other descriptions, so will use those instead.
___
*[api/api-spec.json, line 1250 at r2](https://reviewable.io/reviews/cockroachdb/docs/12323#-MoeE1L2SOmTzNrDoeGj-r2-1250:-Mt59N_yBUn8RDS4Q-Ee:b-vf9b2g) ([raw file](https://github.com/cockroachdb/docs/blob/e17b7956bb93cc3c3a3de8ac5b44594d63e68e10/api/api-spec.json#L1250)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

There has been a mix of either i.e. or e.g. throughout, sometimes followed by a comma, sometimes not. I tend to drop the comma after, but either way might want to just update across the board.
</blockquote></details>

Hmm i've been keeping the comma! Will update throughout.
___
*[api/api-spec.json, line 1541 at r2](https://reviewable.io/reviews/cockroachdb/docs/12323#-MoeE1L2SOmTzNrDoeGj-r2-1541:-Mt5DAes98hwAL1DtBuO:b-f8fm5v) ([raw file](https://github.com/cockroachdb/docs/blob/e17b7956bb93cc3c3a3de8ac5b44594d63e68e10/api/api-spec.json#L1541)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

LeasePreferences --> "lease_preferences"? Wasn't sure if that was referring to line 1545 in this review? Not sure which this was referring to
</blockquote></details>

I think this might correspond to 1551, so I think it'll be ok to leave this.
___
*[api/api-spec.json, line 1546 at r2](https://reviewable.io/reviews/cockroachdb/docs/12323#-MoeE1L2SOmTzNrDoeGj-r2-1546:-Mt5DHmM4_jpmKqJ9XSM:ba3ulst) ([raw file](https://github.com/cockroachdb/docs/blob/e17b7956bb93cc3c3a3de8ac5b44594d63e68e10/api/api-spec.json#L1546)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

"lease preferred" or "least preferred"?
</blockquote></details>

Nice catch!! Fixed.
___
*[api/api-spec.json, line 1554 at r2](https://reviewable.io/reviews/cockroachdb/docs/12323#-MoeE1L2SOmTzNrDoeGj-r2-1554:-Mt5DkGz4nIvVDMI7AfA:bm4u30z) ([raw file](https://github.com/cockroachdb/docs/blob/e17b7956bb93cc3c3a3de8ac5b44594d63e68e10/api/api-spec.json#L1554)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

Here, I wasn't sure about what should be in backticks and what should not be. It seems to depend on whether it's a field or an attribute?
</blockquote></details>

Added backticks to `VoterConstraints` just for consistency within this description.
___
*[api/api-spec.json, line 1611 at r2](https://reviewable.io/reviews/cockroachdb/docs/12323#-MoeE1L2SOmTzNrDoeGj-r2-1611:-Mt5DwOHEPG2SAdgqC-0:bu28u3d) ([raw file](https://github.com/cockroachdb/docs/blob/e17b7956bb93cc3c3a3de8ac5b44594d63e68e10/api/api-spec.json#L1611)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

Does this need a period at the end rather than a comma within the " "? Also this title didn't quite make sense to me.  As in, I wasn't sure what was being indicated if that makes sense. The object level? Maybe "... indicates the object level for objects with a..."
</blockquote></details>

I think this might actually be an incomplete title? @itsbilal can you confirm?
___
*[api/api-spec.json, line 1619 at r2](https://reviewable.io/reviews/cockroachdb/docs/12323#-MoeE1L2SOmTzNrDoeGj-r2-1619:-Mt5E03OEsl7OJQPrKcb:biipln1) ([raw file](https://github.com/cockroachdb/docs/blob/e17b7956bb93cc3c3a3de8ac5b44594d63e68e10/api/api-spec.json#L1619)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

Check on the single quotes vs. double quotes here for target_id
</blockquote></details>

Using backticks as above.
___
*[api/api-spec.json, line 1659 at r2](https://reviewable.io/reviews/cockroachdb/docs/12323#-MoeE1L2SOmTzNrDoeGj-r2-1659:-Mt5E5ZuDyyU34JaG8y3:b-kwg3c3) ([raw file](https://github.com/cockroachdb/docs/blob/e17b7956bb93cc3c3a3de8ac5b44594d63e68e10/api/api-spec.json#L1659)):*
<details><summary><i>Previously, kathancox (Kathryn Hancox) wrote…</i></summary><blockquote>

Same comment as above a few comments, if relevant
</blockquote></details>

@itsbilal should this example be updated to use uppercase letters or dots?

**Comment by taroface** [comment]
@kathancox thanks so much for the detailed review!

**Review by itsbilal — COMMENTED** [review]
Sorry for the late reply! Looks good to me 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/12323#-:-MuNfhy5-c1apOBHldwR:b9mvpul)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @itsbilal, @kathancox, and @mikeCRL)

___
*[api/api-spec.json, line 190 at r2](https://reviewable.io/reviews/cockroachdb/docs/12323#-MoeE1L2SOmTzNrDoeGj-r2-190:-MuNfKrR3eb73jPLDS6J:bsaa7yh) ([raw file](https://github.com/cockroachdb/docs/blob/e17b7956bb93cc3c3a3de8ac5b44594d63e68e10/api/api-spec.json#L190)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

Good point. @itsbilal could you advise on whether the example needs to be updated?
</blockquote></details>

Yeah, the example should probably be updated in code to make this clearer. Until then, you could change the example to have an upper case letter or so. The current example is trying to point out that the space is a sql unsafe character. 
___
*[api/api-spec.json, line 1157 at r2](https://reviewable.io/reviews/cockroachdb/docs/12323#-MoeE1L2SOmTzNrDoeGj-r2-1157:-MuNeQxxEeBIT8n1xfnp:bfh6eqr) ([raw file](https://github.com/cockroachdb/docs/blob/e17b7956bb93cc3c3a3de8ac5b44594d63e68e10/api/api-spec.json#L1157)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

Not sure. @itsbilal can you confirm whether "Subzone" should be singular here?
</blockquote></details>

It can be lowercase as it's referencing to the one in ZoneConfig, but it should be plural as it's a plural property of `ZoneConfig`. Hope that makes sense!
___
*[api/api-spec.json, line 1611 at r2](https://reviewable.io/reviews/cockroachdb/docs/12323#-MoeE1L2SOmTzNrDoeGj-r2-1611:-MuNf0DJ9bpDpaUrhMDf:b-5umlvm) ([raw file](https://github.com/cockroachdb/docs/blob/e17b7956bb93cc3c3a3de8ac5b44594d63e68e10/api/api-spec.json#L1611)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

I think this might actually be an incomplete title? @itsbilal can you confirm?
</blockquote></details>

Yes, this is incomplete and looks like a parsing error. The whole comment that you can copy is:

```
// ZoneConfigurationLevel indicates, for objects with a Zone Configuration,
// the object level at which the configuration is defined. This is needed
// because objects without a specifically indicated Zone Configuration will
// inherit the configuration of their "parent".
```

Valid levels are unknown (0), cluster (1), database (2), or table (3).
___
*[api/api-spec.json, line 1659 at r2](https://reviewable.io/reviews/cockroachdb/docs/12323#-MoeE1L2SOmTzNrDoeGj-r2-1659:-MuNffHv9otGCriGWRI1:bo4btyu) ([raw file](https://github.com/cockroachdb/docs/blob/e17b7956bb93cc3c3a3de8ac5b44594d63e68e10/api/api-spec.json#L1659)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

@itsbilal should this example be updated to use uppercase letters or dots?
</blockquote></details>

Addressed above!

**Review by taroface — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/12323)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @itsbilal, @kathancox, and @mikeCRL)

___
*[api/api-spec.json, line 190 at r2](https://reviewable.io/reviews/cockroachdb/docs/12323#-MoeE1L2SOmTzNrDoeGj-r2-190:-MuRfXxD5n_eRY42431q:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/e17b7956bb93cc3c3a3de8ac5b44594d63e68e10/api/api-spec.json#L190)):*
<details><summary><i>Previously, itsbilal (Bilal Akhtar) wrote…</i></summary><blockquote>

Yeah, the example should probably be updated in code to make this clearer. Until then, you could change the example to have an upper case letter or so. The current example is trying to point out that the space is a sql unsafe character. 
</blockquote></details>

Done.
___
*[api/api-spec.json, line 1157 at r2](https://reviewable.io/reviews/cockroachdb/docs/12323#-MoeE1L2SOmTzNrDoeGj-r2-1157:-MuRfeWp35d_yi6eC9E9:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/e17b7956bb93cc3c3a3de8ac5b44594d63e68e10/api/api-spec.json#L1157)):*
<details><summary><i>Previously, itsbilal (Bilal Akhtar) wrote…</i></summary><blockquote>

It can be lowercase as it's referencing to the one in ZoneConfig, but it should be plural as it's a plural property of `ZoneConfig`. Hope that makes sense!
</blockquote></details>

Done.
___
*[api/api-spec.json, line 1611 at r2](https://reviewable.io/reviews/cockroachdb/docs/12323#-MoeE1L2SOmTzNrDoeGj-r2-1611:-MuRh3NY3e3Y6XMKXFw5:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/e17b7956bb93cc3c3a3de8ac5b44594d63e68e10/api/api-spec.json#L1611)):*
<details><summary><i>Previously, itsbilal (Bilal Akhtar) wrote…</i></summary><blockquote>

Yes, this is incomplete and looks like a parsing error. The whole comment that you can copy is:

```
// ZoneConfigurationLevel indicates, for objects with a Zone Configuration,
// the object level at which the configuration is defined. This is needed
// because objects without a specifically indicated Zone Configuration will
// inherit the configuration of their "parent".
```

Valid levels are unknown (0), cluster (1), database (2), or table (3).
</blockquote></details>

Done.
___
*[api/api-spec.json, line 1659 at r2](https://reviewable.io/reviews/cockroachdb/docs/12323#-MoeE1L2SOmTzNrDoeGj-r2-1659:-MuRh62U1v1w8oRmbVPb:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/e17b7956bb93cc3c3a3de8ac5b44594d63e68e10/api/api-spec.json#L1659)):*
<details><summary><i>Previously, itsbilal (Bilal Akhtar) wrote…</i></summary><blockquote>

Addressed above!
</blockquote></details>

Done.

**Pull request merged**